### PR TITLE
Feat/menu

### PR DESCRIPTION
**Pull request by ProggerOleg**
## Describe your changes
- Created menu layout
- Created rtk request to top-up/withdraw balance

Screenshot from 2023-12-27 21-31-50

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://homecareaid.atlassian.net/jira/software/projects/CC/boards/1?selectedIssue=CC-189)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Comment by Lilly-nn** [comment]
изображение_2023-12-28_154925087
I think menu width should be larger

File: src/components/reusable/header/Menu.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  const handleClose = (): void => {
+    setAnchorEl(null);
+  };
+
+  const handleLogOut = async (): Promise<void> => {
+    try {
+      dispatch(removeToken());
+      dispatch(removeUser());
+      handleClose();
+      window.location.reload();
```
Comment (Lilly-nn) [inline src/components/reusable/header/Menu.tsx:74]: Do we need reload here? Won't be redirecting to sign-in page enough?

Comment (ProggerOleg) [reply src/components/reusable/header/Menu.tsx:74]: Yes, we need it here, there is no rerender until user tries to access anywhere else, and if he presses logout and nothing happens he won't understand, that's why i manually rerender page to direct him to login

**Comment by ProggerOleg** [comment]
> изображение_2023-12-28_154925087 I think menu width should be larger

it's just view from a screenshot from a PC with a ~2000px screen, it's not that small, need to check from default device, but i belive it should be ok

**Comment by Lilly-nn** [comment]
> > изображение_2023-12-28_154925087 I think menu width should be larger
> 
> it's just view from a screenshot from a PC with a ~2000px screen, it's not that small, need to check from default device, but i belive it should be ok

if you checked this, than ok

File: src/components/reusable/header/Menu.tsx (hunk 1 above)
Comment (Lilly-nn) [reply src/components/reusable/header/Menu.tsx:74]: To be honest, I still don't think this is the best approach, but ok

Comment (Simpsonkk) [inline src/components/reusable/header/Menu.tsx:74]: I read prev conversation about reloading. You can just redirect user to login page without reloading.

File: src/components/reusable/header/Menu.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    <>
+      <Button onClick={handleClick} sx={{ padding: '15px' }}>
+        <Arrow className={open ? 'active' : ''}>{children}</Arrow>
+      </Button>
+      <StyledMenu anchorEl={anchorEl} open={open} onClose={handleClose}>
+        <BalanceBlock>
+          <BalanceTitle>{translate('menu.balance')}</BalanceTitle>
+          <BalanceParagraph>
+            <BalanceAmount>
+              <div style={{ opacity: 0.6 }}>$ </div>
```
Comment (Simpsonkk) [inline src/components/reusable/header/Menu.tsx:112]: inline style

**Review by fedasevich — APPROVED** [review]

**Pull request merged**